src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/Record2SolutionMappingTranslatorImpl.java

```diff
@@ -0,0 +1,129 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.engine.binding.BindingBuilder;
+import se.liu.ida.hefquin.engine.data.SolutionMapping;
+import se.liu.ida.hefquin.engine.data.impl.SolutionMappingImpl;
+import se.liu.ida.hefquin.engine.utils.Pair;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.LPG2RDFConfiguration;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.Record2SolutionMappingTranslator;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.RecordEntry;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.TableRecord;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.Value;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.impl.*;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.query.CypherQuery;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.query.CypherVar;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.utils.CypherUtils;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class Record2SolutionMappingTranslatorImpl implements Record2SolutionMappingTranslator {
+
+    @Override
+    public List<SolutionMapping> translateRecord(final TableRecord record, final LPG2RDFConfiguration conf,
+                                                 final CypherQuery query, final Map<CypherVar, Node> varMap) {
+        final List<Pair<CypherVar, ListValue>> lists = new ArrayList<>();
+        SolutionMapping base = translateRecordsPriv(record, conf, query, varMap, lists);
+        return manageLists(base, lists, conf, query, varMap);
+    }
+
+    private List<SolutionMapping> manageLists(final SolutionMapping base, final List<Pair<CypherVar, ListValue>> lists,
```

Comment:
```suggestion
    protected List<SolutionMapping> manageLists(final SolutionMapping base, final List<Pair<CypherVar, ListValue>> lists,
```

Reply:
Additionally, please include a comment that summarizes the purpose of this method.